server/admin_management/tests/routers/test_router_admin_user_access.py

```diff
@@ -0,0 +1,371 @@
+import logging
+from http import HTTPStatus
+
+import starlette.testclient
+import tests.jwt_utils as jwt_utils
+from api.app.constants import AdminRoleAuthGroup, AppEnv, RoleType
+from api.app.main import apiPrefix
+from tests.constants import (TEST_APP_FOM_NAME,
+                             TEST_APP_ROLE_NAME_FOM_REVIEWER,
+                             TEST_APP_ROLE_NAME_FOM_SUBMITTER,
+                             TEST_APPLICATION_ID_FAM,
+                             TEST_APPLICATION_ID_FOM_DEV,
+                             TEST_APPLICATION_ID_FOM_TEST,
+                             TEST_APPLICATION_NAME_FAM,
+                             TEST_DUMMY_COGNITO_USER_ID,
+                             TEST_FOM_DEV_REVIEWER_ROLE_ID,
+                             TEST_FOM_DEV_SUBMITTER_ROLE_ID,
+                             TEST_FOM_TEST_REVIEWER_ROLE_ID,
+                             TEST_FOM_TEST_SUBMITTER_ROLE_ID,
+                             TEST_FOREST_CLIENT_NUMBER,
+                             TEST_FOREST_CLIENT_NUMBER_TWO,
+                             TEST_NEW_BCEID_USER, TEST_NEW_IDIR_USER)
+
+LOGGER = logging.getLogger(__name__)
+test_end_point = f"{apiPrefix}/admin-user-accesses"
+
+# Below tests are based on endpoint json sample output at confluence design:
+# https://apps.nrs.gov.bc.ca/int/confluence/display/FSAST1/Delegated+Access+Administration+Design
```

Comment:
Hi Ian, I wonder if you want to update this confluence page to make the json sample up to date? For example, in the below tests, for app admin, roles will have forest client number None, `assert fom_dev_roles_grants_submitter["forest_clients"] is None`, but it's not shown in the json sample. Think the json example in your last pr is the latest https://github.com/bcgov/nr-forests-access-management/pull/1170
<img width="551" alt="Screen Shot 2024-02-12 at 4 06 48 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/9c87be32-230d-4967-bca1-f397df6f58be">
<img width="580" alt="Screen Shot 2024-02-12 at 4 06 58 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/190dbd95-8776-4250-8f4a-8a1febb57cdf">

Reply:
Good suggestion. I updated the confluence.

However, on the side note, it actually has a layer of translation between python (FastAPI) and Swagger before outputting the swagger return json we see on api's doc (on swagger). I noticed using Python's "Optional" and "Union" types can have difference at the swagger's translation (for example, difference between: None becomes "null" or None becomes => the key/alue being removed). If we do need to make it: None becomes => the key/alue being removed, we can have another ticket to adjust (but not clear what to adjust, the python type, or is there existing swagger conversion configuration for this?).

Reply:
I would prefer to make it optional, so if "None" then remove key/value pair

Reply:
Thanks Catherine.
I also prefer "Optional" than "Union" in this case.
However, it is the "Optional" that Swagger is producing the "null", not removing the key/value pair.
This is the issue between FastAPI(python) and Swagger, not really in our code.
I will create a "nice-to-have tech debt" ticket so we can research/investigate if there is a way to tweak FastAPI for this at current or higher version.